schema-engine/connectors/sql-schema-connector/src/sql_doc_parser.rs

```diff
@@ -0,0 +1,788 @@
+        .strip_prefix_char('$')
+        .map(|input| input.take_until_pattern_or_eol(&[':', ' ']))
+    {
+        match param_pos.parse::<usize>().map_err(|_| {
+            build_error(
+                input,
+                &format!("invalid position. Expected a number found: {param_pos}"),
+            )
+        }) {
+            Ok(param_pos) => Ok((param_input, Some(param_pos))),
+            Err(err) => Err(err),
+        }
+    } else {
+        Ok((input, None))
+    }
+}
+
+fn parse_alias_opt(input: Input<'_>) -> ConnectorResult<(Input<'_>, Option<&'_ str>)> {
+    if let Some((input, alias)) = input
+        .trim_start()
+        .strip_prefix_char(':')
+        .map(|input| input.take_until_pattern_or_eol(&[' ']))
+    {
+        Ok((input, Some(alias)))
+    } else {
+        Ok((input, None))
+    }
+}
+
+fn parse_rest(input: Input<'_>) -> ConnectorResult<Option<&str>> {
+    let input = input.trim_start();
+
+    if input.is_empty() {
+        return Ok(None);
+    }
+
+    Ok(Some(input.trim_end().inner()))
+}
+
+fn validate_param(param: &ParsedParameterDoc<'_>, input: Input<'_>) -> ConnectorResult<()> {
+    if param.is_empty() {
+        return Err(build_error(input, "invalid parameter: could not parse any information"));
+    }
+
+    if param.position.is_none() && param.alias().is_none() {
+        return Err(build_error(input, "missing position or alias (eg: $1:alias)"));
+    }
+
+    Ok(())
+}
+
+fn parse_param(param_input: Input<'_>) -> ConnectorResult<ParsedParameterDoc<'_>> {
+    let input = param_input.strip_prefix_str("@param").unwrap().trim_start();
+
+    let (input, typ) = parse_typ_opt(input)?;
+    let (input, position) = parse_position_opt(input)?;
+    let (input, alias) = parse_alias_opt(input)?;
+    let documentation = parse_rest(input)?;
+
+    let mut param = ParsedParameterDoc::default();
+
+    param.set_typ(typ);
+    param.set_position(position);
+    param.set_alias(alias);
+    param.set_documentation(documentation);
+
+    validate_param(&param, param_input)?;
+
+    Ok(param)
+}
+
+fn parse_description(input: Input<'_>) -> ConnectorResult<Option<&str>> {
+    let input = input.strip_prefix_str("@description").unwrap();
+
+    parse_rest(input)
+}
+
+pub(crate) fn parse_sql_doc(sql: &str) -> ConnectorResult<ParsedSqlDoc<'_>> {
+    let mut parsed_sql = ParsedSqlDoc::default();
+
+    let lines = sql.lines();
+
+    for line in lines {
+        let input = Input(line.trim());
+
+        if let Some(input) = input.strip_prefix_str("--") {
+            let input = input.trim_start();
+
+            if input.starts_with("@description") {
+                parsed_sql.set_description(parse_description(input)?);
+            } else if input.starts_with("@param") {
+                parsed_sql
+                    .add_parameter(parse_param(input)?)
+                    .map_err(|err| build_error(input, err.message().unwrap()))?;
+            }
+        }
+    }
+
+    Ok(parsed_sql)
+}
```

Comment:
As clarified in the call, this logic only handles SQL comments à la:

```sql
-- first comment line
-- second comment line
```

Multi-line comments, like

```sql
/**
 * First comment
 * Second comment
 */
```

are not supported. This is expected as it makes the parsing logic much simpler, and while it may be revisited in the future, we should just document this limitation for the time being.

Reply:
Should we maybe have a new test that showcases failures in the multiline `/** ... */` case?